lib/ddtrace/workers/loop.rb

```diff
@@ -20,7 +20,9 @@ def perform(*args)
 
       def stop_loop
         mutex.synchronize do
-          return false unless run_loop?
+          # Ensure we always inform that a stop was requested.
+          # This guarantees correct shutdown even if the loop
+          # has asynchronously not stated yet.
```

Comment:
This was a race condition when `perform_loop` (just below) is about to be started, but hasn't yet. In this case, `run_loop?` will return false, and we'll never mark this worker as stopped.